ratpack-bearer-auth/src/main/java/st/ratpack/auth/handler/CookieAuthHandler.java

```diff
@@ -0,0 +1,71 @@
+package st.ratpack.auth.handler;
+
+import io.netty.handler.codec.http.cookie.Cookie;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ratpack.exec.Promise;
+import ratpack.handling.Context;
+import ratpack.handling.Handler;
+import ratpack.registry.Registry;
+import st.ratpack.auth.DefaultUser;
+import st.ratpack.auth.OAuthToken;
+import st.ratpack.auth.TokenValidator;
+import st.ratpack.auth.User;
+
+import java.util.Optional;
+
+public class CookieAuthHandler implements Handler {
+	private static final Logger LOG = LoggerFactory.getLogger(CookieAuthHandler.class);
+	private final TokenValidator validator;
+	private final String tokenCookieName;
+
+	public CookieAuthHandler(
+		String tokenCookieName,
+		TokenValidator validator
+	) {
+		this.tokenCookieName = tokenCookieName;
+		this.validator = validator;
+	}
+
+	@Override
+	public void handle(Context ctx) throws Exception {
+		Optional<Cookie> authCookie = ctx.getRequest()
+			.getCookies()
+			.stream()
+			.filter(cookie -> cookie.name().equals(tokenCookieName))
+			.findFirst();
+
+		if (authCookie.isPresent()) {
+			LOG.info("Found " + tokenCookieName + " cookie and now attempting to validate.");
+			Promise<Optional<OAuthToken>> optionalToken = validator.validate(authCookie.get().value());
+			optionalToken
+				.onError(throwable -> {
+					LOG.error("Error validating " + tokenCookieName + " cookie.", throwable);
+					ctx.next();
+				})
+				.then((oAuthToken) -> {
+					if (oAuthToken.isPresent()) {
+						LOG.info(tokenCookieName + " cookie is valid.");
+						ctx.next(Registry.of(registrySpec -> {
+							OAuthToken authToken = oAuthToken.get();
+							// Add the oauth token object to the registry
+							registrySpec.add(authToken);
+
+							if (authToken.isUserToken()) {
+								DefaultUser.Builder  builder = new DefaultUser.Builder(authToken);
+								User user = builder.build();
+								// Add the user to the registry
+								registrySpec.add(user);
+							}
+						}));
+					} else {
+						LOG.info(tokenCookieName + " cookie is not valid.");
```

Comment:
This probably isn't useful unless we log the cookie as well.  Otherwise its impossible to debug why the cookie is invalid if we see a lot of this.